src/mono/mono.proj

```diff
@@ -581,6 +581,8 @@
     </ItemGroup>
     <ItemGroup>
       <_MonoCMakeArgs Include="-DMONO_COMPONENTS_RID=$(TargetOS)-$(TargetArchitecture)" />
+      <_MonoCMakeArgs Condition="'$(TargetArchitecture)' == 'wasm' or '$(TargetArchitecture)' == 'wasm32'" Include="-DCLR_CMAKE_HOST_ARCH=$(BuildArchitecture)" />
```

Comment:
we don't use `wasm32` as `TargetArchitecture` in dotnet/runtime, we use `wasm` to mean the same thing.